src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-list/cephfs-subvolume-list.component.ts

```diff
@@ -174,16 +186,34 @@ export class CephfsSubvolumeListComponent implements OnInit, OnChanges {
   }
 
   removeSubVolumeModal() {
-    const name = this.selection.first().name;
-    this.modalService.show(CriticalConfirmationModalComponent, {
-      itemDescription: 'subvolume',
-      itemNames: [name],
-      actionDescription: 'remove',
-      submitActionObservable: () =>
-        this.taskWrapper.wrapTaskAroundCall({
-          task: new FinishedTask('cephfs/subvolume/remove', { subVolumeName: name }),
-          call: this.cephfsSubVolume.remove(this.fsName, name)
-        })
+    this.removeForm = new CdFormGroup({
+      retainSnapshots: new FormControl(false)
+    });
+    this.errorMessage = '';
+    this.selectedName = this.selection.first().name;
+    this.modalRef = this.modalService.show(CriticalConfirmationModalComponent, {
+      actionDescription: 'Remove',
+      itemNames: [this.selectedName],
+      itemDescription: 'Subvolume',
+      childFormGroup: this.removeForm,
+      childFormGroupTemplate: this.removeTmpl,
+      submitAction: () =>
+        this.taskWrapper
+          .wrapTaskAroundCall({
+            task: new FinishedTask('cephfs/subvolume/remove', { subVolumeName: this.selectedName }),
+            call: this.cephfsSubVolume.remove(
+              this.fsName,
+              this.selectedName,
+              this.removeForm.getValue('retainSnapshots')
+            )
+          })
+          .subscribe({
+            complete: () => this.modalRef.close(),
+            error: (error) => {
+              this.modalRef.componentInstance.stopLoadingSpinner();
+              this.errorMessage = error.error['traceback'].split('RuntimeError: ').at(-1);
```

Comment:
If the `DashboardException` is properly raised from the API, then we would only need to store the error detail.
```suggestion
              this.errorMessage = error.error.detail;
```